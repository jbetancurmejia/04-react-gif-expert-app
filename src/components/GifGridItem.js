
import PropTypes from 'prop-types';
import React, { Component } from 'react';


export const GifGridItem = ({title,url}) => {

  
  return <div className="card animate__animated animate__bounce animate__delay-2s">
   <img  src={url}  alt={title}  />
   <p>{title}</p>   
  </div>;
};

GifGridItem.prototypes ={  
  title:PropTypes.string.isRequiered,
  url:PropTypes.string.IsRequiered
  
}

