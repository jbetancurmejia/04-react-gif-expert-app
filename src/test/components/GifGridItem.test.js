import React from 'react';
import {shallow} from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en <GifGridItem />', () => {


    const title="prueba";
    const url="http://prueba.com"
    const wrapper= shallow( <GifGridItem title={title} url={url} /> )
    
    test('debe de mostrar correctamente el componente', () => {

       expect ( wrapper ).toMatchSnapshot();

    });   
    


    test('debe tener un pÃ¡rrafo con el title', () => {
        const p = wrapper.find('p');
        expect( p.text().trim()).toBe(title);

    });

    test('debe tener la imagen igual al url y alt de los props', () => {
        const img = wrapper.find('img');
        // console.log(img.props().alt)
        expect( img.props().alt).toBe(title);
        expect( img.props().src).toBe(url);

    });

    test('Debe tener la clase animate__delay', () => {
      
        const div = wrapper.find('div');
        const className=  div.props().className;
        expect(className.includes('animate__animated')).toBe(true);
        //expect( img.props().src).toBe(url);
    });
    
    

});
