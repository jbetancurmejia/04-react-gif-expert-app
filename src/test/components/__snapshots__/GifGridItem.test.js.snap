// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> debe de mostrar correctamente el componente 1`] = `
<div
  className="card animate__animated animate__bounce animate__delay-2s"
>
  <img
    alt="prueba"
    src="http://prueba.com"
  />
  <p>
    prueba
  </p>
</div>
`;
